fog/ingest/server/src/controller.rs

```diff
@@ -716,11 +716,11 @@ where
                         .map(|x| x + 1)
                         .unwrap_or(0),
                 );
-                if !self.recovery_db.new_ingress_key(&our_pubkey, start_block)? {
-                    return Err(PeerBackupError::CreatingNewIngressKey.into());
-                };
 
-                start_block
+                match self.recovery_db.new_ingress_key(&our_pubkey, start_block) {
+                    Ok(accepted_start_block) => accepted_start_block,
+                    Err(err) => return Err(PeerBackupError::CreatingNewIngressKey.into()),
```

Comment:
Note that this line slightly modifies the behavior of how we reported errors from the `new_ingress_key` method. Previously, we used the `?` operator to pass along the two errors ([here](https://github.com/mobilecoinfoundation/mobilecoin/blob/25aad6e0ffef1f00e7fde8a161c7c7e72c4a9b74/fog/sql_recovery_db/src/lib.rs#L198) and [here](https://github.com/mobilecoinfoundation/mobilecoin/blob/25aad6e0ffef1f00e7fde8a161c7c7e72c4a9b74/fog/sql_recovery_db/src/lib.rs#L210)) reported by the method, both of which are r2d2 Errors. It would then pass a `PeerBackupError::CreatingNewIngressKey` when the key wasn't successfully inserted.

Instead of using a boolean to signify a failed key insertion, the code now emits a newly-defined `IngressKeyUnsuccessfulInsert` defined the sql_db crate. This method then takes any error emitted by the `new_ingress_key` method and returns `PeerBackupError::CreatingNewIngressKey`. So it will no longer pass along those r2d2 Errors. 

I investigated the usage of these errors, and couldn't see any specific use cases for the error type emitted here. Therefore, I think it's safe to make this small change in error handling, but just wanted to call it out here in case I'm missing something.

Reply:
it looks good to me, thank you